layer/executables/entrypoint/dataset.py

```diff
@@ -34,84 +32,74 @@
 
 
 def _run(
-    dataset_definition: FunctionDefinition, config: Config, fabric: Fabric, run_id: str
+    dataset_definition: FunctionDefinition,
+    client: LayerClient,
+    tracker: RunProgressTracker,
+    fabric: Fabric,
+    run_id: str,
 ) -> None:
-
-    with LayerClient(config.client, logger).init() as client:
-
-        progress_tracker = get_progress_tracker(
-            url=config.url,
-            project_name=dataset_definition.project_name,
-            account_name=dataset_definition.account_name,
-        )
-
-        with progress_tracker.track() as tracker:
-            tracker.add_asset(AssetType.DATASET, dataset_definition.asset_name)
-
-            _register_function(
-                client, dataset=dataset_definition, tracker=tracker, fabric=fabric
+    _register_function(
+        client, dataset=dataset_definition, tracker=tracker, fabric=fabric
+    )
+    tracker.mark_dataset_building(dataset_definition.asset_name)
+    # TODO pass path to API instead
+    current_project_uuid = verify_project_exists_and_retrieve_project_id(
+        client, dataset_definition.project_full_name
+    )
+    dataset_build_id = client.data_catalog.initiate_build(
+        current_project_uuid,
+        dataset_definition.asset_name,
+        fabric.value,
+    )
```

Comment:
since `client` holds specialized clients, we _could_ move TODOs there and simplify the code here:
```python
client.initiate_dataset_build(dataset_definition.asset_path)
```
and we do this inside `client`
```python
   # TODO pass path to API instead
    current_project_uuid = verify_project_exists_and_retrieve_project_id(
        client, dataset_definition.project_full_name
    )
    dataset_build_id = client.data_catalog.initiate_build(
        current_project_uuid,
        dataset_definition.asset_name,
        fabric.value,
    )
```

Reply:
agreed :) noting for a subsequent refactor PR